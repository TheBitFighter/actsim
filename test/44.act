defproc test()
{
  int<64> emm1;
  chan(int<64>) EMM1;
  int<64> emm2;

  chp {
	emm1 := 0x0101010110101010;
        EMM1!emm1,EMM1?emm2;
	log ("emm2_high: ", emm2{63..32}, " emm2_low: ", emm2{31..0} );
	log ("emm1: ", emm1{63..0});
	EMM1!emm1{63..0}, EMM1?emm2;
	log ("emm2_high: ", emm2{63..32}, " emm2_low: ", emm2{31..0} );
	EMM1!{emm1{63},emm1{62..0}}, EMM1?emm2;
	log ("emm2_high: ", emm2{63..32}, " emm2_low: ", emm2{31..0} )
  }
}
