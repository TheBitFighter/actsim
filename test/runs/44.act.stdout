[      20 t#:0] <> emm2_high: 16843009 emm2_low: 269488144
[      20 t#:0] <> emm1: 72340173090721808
[      30 t#:0] <> emm2_high: 16843009 emm2_low: 269488144
[      40 t#:0] <> emm2_high: 16843009 emm2_low: 269488144
